go.mod

```diff
@@ -9,9 +9,12 @@ require (
 	github.com/alecthomas/jsonschema v0.0.0-20220216202328-9eeeec9d044b
 	github.com/benbjohnson/clock v1.3.5
 	github.com/celestiaorg/celestia-app v1.13.0
+	github.com/celestiaorg/celestia-app/v2 v2.0.0
 	github.com/celestiaorg/go-fraud v0.2.1
 	github.com/celestiaorg/go-header v0.6.2
 	github.com/celestiaorg/go-libp2p-messenger v0.2.0
+	github.com/celestiaorg/go-square v1.1.0
+	github.com/celestiaorg/go-square/merkle v0.0.0-20240117232118-fd78256df076
```

Comment:
Looks like `merkle` package got removed here https://github.com/celestiaorg/go-square/pull/91/files

Any good reason why we import it?

Reply:
That PR doesn't look like it applied to go-square v1.x which is what is currently used by celestia-app v2.x. 

```
$ go mod why github.com/celestiaorg/go-square/merkle
# github.com/celestiaorg/go-square/merkle
github.com/celestiaorg/celestia-node/blob
github.com/celestiaorg/celestia-app/v2/pkg/proof
github.com/celestiaorg/go-square/merkle
```

TBH I'm not sure why it is listed as a direct dependency instead of an indirect dependency b/c I can't see an explicit import for it in this repo.

Reply:
Can you run `go mod tidy` please?

Reply:
I tried. I also manually removed this dependency, ran `go mod tidy` and it was added back.

Reply:
Yeah, looks like it should be imported, I read `go mod why ...` incorrectly. Resolving, thanks.